samwise.js

```diff
@@ -1,31 +1,28 @@
-/**
```

Comment:
SCRIPT-This block of code could benefit from some comments explaining the logic behind it.